program/tests/close_account.rs

```diff
@@ -1,200 +1,93 @@
 
-    let space = Mint::LEN;
-    let tx = Transaction::new_signed_with_payer(
-        &[
-            system_instruction::create_account(
-                &context.payer.pubkey(),
-                &mint.pubkey(),
-                rent.minimum_balance(space),
-                space as u64,
-                token_program_id,
-            ),
-            instruction::initialize_mint(
-                token_program_id,
-                &mint.pubkey(),
-                &mint_authority_pubkey,
-                None,
-                9,
-            )
-            .unwrap(),
-        ],
-        Some(&context.payer.pubkey()),
-        &[&context.payer, mint],
-        context.last_blockhash,
-    );
-    context.banks_client.process_transaction(tx).await.unwrap();
-    let space = Account::LEN;
-    let tx = Transaction::new_signed_with_payer(
-        &[
-            system_instruction::create_account(
-                &context.payer.pubkey(),
-                &token_account.pubkey(),
-                rent.minimum_balance(space),
-                space as u64,
-                token_program_id,
-            ),
-            instruction::initialize_account(
-                token_program_id,
-                &token_account.pubkey(),
-                &mint.pubkey(),
-                owner,
-            )
-            .unwrap(),
-        ],
-        Some(&context.payer.pubkey()),
-        &[&context.payer, token_account],
-        context.last_blockhash,
-    );
-    context.banks_client.process_transaction(tx).await.unwrap();
-}
+    let owner = Pubkey::new_unique();
+    let mint = Pubkey::new_unique();
+    let account = Pubkey::new_unique();
+    let decimals = 9;
 
-#[tokio::test]
-async fn success_init_after_close_account() {
-    let program_test =
-        ProgramTest::new("spl_token", spl_token::id(), processor!(Processor::process));
-    let mut context = program_test.start_with_context().await;
-    let mint = Keypair::new();
-    let token_account = Keypair::new();
-    let owner = Keypair::new();
-    let token_program_id = spl_token::id();
-    setup_mint_and_account(
-        &mut context,
-        &mint,
-        &token_account,
-        &owner.pubkey(),
-        &token_program_id,
-    )
-    .await;
+    let owner_account = AccountSharedData::new(1_000_000_000, 0, &system_program::id());
+    let mint_account = setup::setup_mint_account(None, None, 0, decimals);
+    let token_account = setup::setup_token_account(&mint, &owner, 0);
 
-    let destination = Pubkey::new_unique();
-    let tx = Transaction::new_signed_with_payer(
+    mollusk.process_and_validate_instruction_chain(
         &[
-            instruction::close_account(
-                &token_program_id,
-                &token_account.pubkey(),
-                &destination,
-                &owner.pubkey(),
-                &[],
-            )
-            .unwrap(),
+            instruction::close_account(&spl_token::id(), &account, &owner, &owner, &[]).unwrap(),
```

Comment:
The previous version of this test had a few more checks -- can this retain the `destination` account that receives the lamports from close and check that `destination` has lamports?

Reply:
Done!